api/server/server.go

```diff
@@ -886,17 +886,28 @@ func (s *Server) getImagesGet(eng *engine.Engine, version version.Version, w htt
 	if err := parseForm(r); err != nil {
 		return err
 	}
-	if version.GreaterThan("1.0") {
+
+	useJSON := version.GreaterThan("1.0")
+	if useJSON {
 		w.Header().Set("Content-Type", "application/x-tar")
 	}
-	var job *engine.Job
+
+	imageExportConfig := &graph.ImageExportConfig{
+		Engine:    eng,
+		Outstream: utils.NewWriteFlusher(w),
+	}
 	if name, ok := vars["name"]; ok {
-		job = eng.Job("image_export", name)
+		imageExportConfig.Names = []string{name}
 	} else {
-		job = eng.Job("image_export", r.Form["names"]...)
+		imageExportConfig.Names = r.Form["names"]
 	}
-	job.Stdout.Add(w)
-	return job.Run()
+
+	if err := s.daemon.Repositories().ImageExport(imageExportConfig); err != nil {
+		sf := streamformatter.NewStreamFormatter(useJSON)
```

Comment:
Why we just don't return error here? I'm pretty sure that you're not returning `JSONError` from `ImageExport`.